spring-amqp/src/main/java/org/springframework/amqp/support/postprocessor/AbstractCompressingPostProcessor.java

```diff
@@ -74,13 +78,19 @@ public Message postProcessMessage(Message message) throws AmqpException {
 		try {
 			OutputStream zipper = getCompressorStream(zipped);
 			FileCopyUtils.copy(new ByteArrayInputStream(message.getBody()), zipper);
-			MessageProperties messageProperties = message.getMessageProperties();
-			String currentEncoding = messageProperties.getContentEncoding();
-			messageProperties
-					.setContentEncoding(getEncoding() + (currentEncoding == null ? "" : ":" + currentEncoding));
+			MessageProperties originalProperties = message.getMessageProperties();
+			MessageBuilderSupport<MessageProperties> messagePropertiesMessageBuilder =
```

Comment:
Why the abstract type here instead of `MessagePropertiesBuilder` ?

Reply:
That is what returned by the `setContentEncoding()`.

I guess I can rework the logic here into the `MessageBuilder` instead...